File: daily-coding-challenge/2025/august/py/space_jam.py
"""

S P A C E J A M

Given a string, remove all spaces from the string, insert two spaces between every character, convert all alphabetical letters to uppercase, and return the result.
  Non-alphabetical characters should remain unchanged (except for spaces).

Tests:
  space_jam("freeCodeCamp") should return "F  R  E  E  C  O  D  E  C  A  M  P".
  space_jam("   free   Code   Camp   ") should return "F  R  E  E  C  O  D  E  C  A  M  P".
  space_jam("Hello World?!") should return "H  E  L  L  O  W  O  R  L  D  ?  !".
  space_jam("C@t$ & D0g$") should return "C  @  T  $  &  D  0  G  $".
  space_jam("allyourbase") should return "A  L  L  Y  O  U  R  B  A  S  E".

"""


def space_jam(s):
    return "  ".join(list(s.replace(" ", "").upper()))


print(space_jam("freeCodeCamp"))  # "F  R  E  E  C  O  D  E  C  A  M  P"
print(space_jam("   free   Code   Camp   "))  # "F  R  E  E  C  O  D  E  C  A  M  P"
print(space_jam("Hello World?!"))  # "H  E  L  L  O  W  O  R  L  D  ?  !"
print(space_jam("C@t$ & D0g$"))  # "C  @  T  $  &  D  0  G  $"
print(space_jam("allyourbase"))  # "A  L  L  Y  O  U  R  B  A  S  E"

File: daily-coding-challenge/2025/august/js/spaceJam.js
/*

S P A C E J A M

Given a string, remove all spaces from the string, insert two spaces between every character, convert all alphabetical letters to uppercase, and return the result.
  Non-alphabetical characters should remain unchanged (except for spaces).

Tests:
  spaceJam("freeCodeCamp") should return "F  R  E  E  C  O  D  E  C  A  M  P".
  spaceJam("   free   Code   Camp   ") should return "F  R  E  E  C  O  D  E  C  A  M  P".
  spaceJam("Hello World?!") should return "H  E  L  L  O  W  O  R  L  D  ?  !".
  spaceJam("C@t$ & D0g$") should return "C  @  T  $  &  D  0  G  $".
  spaceJam("allyourbase") should return "A  L  L  Y  O  U  R  B  A  S  E".

*/

function spaceJam(s) {
  return s.replaceAll(" ", "").toUpperCase().split("").join("  ");
}

console.log(spaceJam("freeCodeCamp")); // "F  R  E  E  C  O  D  E  C  A  M  P"
console.log(spaceJam("   free   Code   Camp   ")); // "F  R  E  E  C  O  D  E  C  A  M  P"
console.log(spaceJam("Hello World?!")); // "H  E  L  L  O  W  O  R  L  D  ?  !"
console.log(spaceJam("C@t$ & D0g$")); // "C  @  T  $  &  D  0  G  $"
console.log(spaceJam("allyourbase")); // "A  L  L  Y  O  U  R  B  A  S  E"
